File: src/main/java/com/github/njuro/jard/user/dto/CurrentUserPasswordEditDto.java
package com.github.njuro.jard.user.dto;

import static com.github.njuro.jard.common.InputConstraints.MIN_PASSWORD_LENGTH;

import com.github.njuro.jard.user.User;
import javax.validation.constraints.AssertTrue;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/** DTO for changing current user's password */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@SuppressWarnings("JavadocReference")
public class CurrentUserPasswordEditDto {

  /** {@link User#password} */
  @NotNull private String currentPassword;

  /** {@link User#password} */
  @Size(min = MIN_PASSWORD_LENGTH, message = "{validation.user.password.length}")
  private String newPassword;

  /** {@link User#password} */
  private String newPasswordRepeated;

  /** Validates that {@link #newPassword} and {@link #newPasswordRepeated} are equal. */
  @AssertTrue(message = "{validation.user.password.match}")
  public boolean isPasswordMatching() {
    return newPassword != null && newPassword.equals(newPasswordRepeated);
  }
}

File: src/main/kotlin/com/github/njuro/jard/rewrite/user/dto/CurrentUserPasswordEditDto.kt
package com.github.njuro.jard.rewrite.user.dto

import com.github.njuro.jard.common.InputConstraints
import javax.validation.constraints.AssertTrue
import javax.validation.constraints.Size

/** DTO for changing current user's password  */
data class CurrentUserPasswordEditDto(
    /** [User.password]  */
    val currentPassword: String,

    /** [User.password]  */
    @Size(
        min = InputConstraints.MIN_PASSWORD_LENGTH,
        message = "{validation.user.password.length}"
    )
    val newPassword: String,

    /** [User.password]  */
    val newPasswordRepeated: String
) {
    /** Validates that [.newPassword] and [.newPasswordRepeated] are equal.  */
    @AssertTrue(message = "{validation.user.password.match}")
    fun isPasswordMatching(): Boolean = newPassword == newPasswordRepeated
}
